packages/formik/src/Formik.tsx

```diff
@@ -504,35 +510,37 @@ export function useFormik<Values extends FormikValues = FormikValues>({
             value: maybePromise as string | undefined,
           },
         });
-        return Promise.resolve(maybePromise as string | undefined);
+        return Promise.resolve(maybePromise);
       }
     } else if (props.validationSchema) {
       dispatch({ type: 'SET_ISVALIDATING', payload: true });
-      return runValidationSchema(state.values, name)
-        .then((x: any) => x)
```

Comment:
this is not a type change, but removed as I thought it was left behind in a refactor. 
Maybe it's to avoid a race-condition?

Reply:
let's keep this a typescript-only change